File: ruby-neo4j-driver/friends_of_a_friend.rb
require './neo4j_init'

# Get session
session = Neo4jInit.getSession("../config.yml")

# Friends of a friend
foaf_query = <<QUERY
  MATCH (person:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf)
  WHERE person.name = {name}
    AND NOT (person)-[:KNOWS]-(foaf)
  RETURN foaf.name AS name
QUERY

response = session.query(foaf_query, name: 'Joe')
puts response.map(&:name)
puts '---------------------'

File: elixir-neo4j-driver/social/friends_of_a_friend.exs
[[{'username', username}, {'password', password}, {'host', host}, _ , _]] = :yamerl_constr.file("../../config.yml")
{:ok, _neo} = Bolt.Sips.start_link(url: "bolt://#{username}:#{password}@#{host}")
conn = Bolt.Sips.conn()

# common friends
foaf = "Joe"
common_friends_query = """
  MATCH (person:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf)
  WHERE person.name = \"#{foaf}\"
  AND NOT (person)-[:KNOWS]-(foaf)
  RETURN foaf.name AS name
"""

Bolt.Sips.query!(conn, common_friends_query) |> IO.inspect
